File: A2/A2-023/A2-023.cpp
#include<iostream>
using namespace std;

int consecA = 0, maxConsecA = 0;
bool isUnknown = true, isRabbit = true;
string txt;

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);
    
    // Input string and make it all lowercase
    cin >> txt;
    for(int i = 0; i < txt.size(); i++)
    {
        txt[i] = tolower(txt[i]);
    }

    // Consider string
    for(int i = 0; i < txt.size(); i++)
    {
        bool c1, c2, c3;
        
        // A character that isn't 'i' and 't' detected
        c1 = txt[i] != 'i';
        c2 = txt[i] != 't';
        if(isUnknown && c1 && c2) { isUnknown = false; }
        
        // After 'r' must be 'a'
        if(txt[i] == 'r')
        {
            c1 = (i + 1 == txt.size()); // 'r' is the last character
            c2 = (txt[i + 1] != 'a');   // After 'r' is not 'a'
            
            if(c1 || c2)
            {
                if(c1) { cout << "no " << i; }
                else { cout << "no " << i + 1; }
                
                isRabbit = false;
                break;
            }
        }

        // Before 'a' must be 'r' or 'a'
        if(txt[i] == 'a')
        {
            consecA++;  // Count consecutive 'a'

            c1 = (i == 0);             // 'a' is the first character
            c2 = (txt[i - 1] != 'r');  // Before 'a' is not 'r'
            c3 = (txt[i - 1] != 'a');  // Before 'a' is not 'a'

            if(c1 || (c2 && c3))
            {
                cout << "no " << i << '\n';
                isRabbit = false;
                break;
            }
        }
        // Reset consecutive 'a' counter
        else
        {
            maxConsecA = max(maxConsecA, consecA);
            consecA = 0;
        }

        // After 'b' must be 'i' or 't'
        if(txt[i] == 'b')
        {
            c1 = (i + 1 == txt.size()); // 'b' is the last character
            c2 = (txt[i + 1] != 'i');   // After 'b' is not 'i'
            c3 = (txt[i + 1] != 't');   // After 'b' is not 't'

            if(c1 || (c2 && c3))
            {
                if(c1) { cout << "no " << i; }
                else { cout << "no " << i + 1; }

                isRabbit = false;
                break;
            }
        }
    }
    // Don't forget to find maximum consecutive once again
    maxConsecA = max(maxConsecA, consecA);

    // Output
    if(isUnknown) { cout << "unknown " << txt.size(); }
    else if(isRabbit) { cout << "yes " << maxConsecA; }
    return 0;
}

File: A2/A2-023/A2-023.py
# Input string and made it all lowercase
txt = input().strip().lower()

# Consider string
consecutive_a = 0
max_consecutive_a = 0
is_unknown = True
is_rabbit = True

for i in range(len(txt)):
    # A character that isn't 'i' and 't' detected
    if(is_unknown and txt[i] not in ['i', 't']):
        is_unknown = False

    # After 'r' must be 'a'
    if(txt[i] == 'r'):
        # 'r' is the last character or after 'r' is not 'a'
        if(i + 1 == len(txt) or txt[i + 1] != 'a'):
            if(i + 1 == len(txt)):
                print("no", i)
            else:
                print("no", i + 1)
            is_rabbit = False
            break

    # Before 'a' must be 'r' or 'a'
    if(txt[i] == 'a'):
        consecutive_a += 1  # Count consecutive 'a'
        # 'a' is the first character or before 'a' is not 'r' or 'a'
        if(i == 0 or txt[i - 1] not in ['r', 'a']):
            print("no", i)
            is_rabbit = False
            break
    # Reset consecutive 'a' counter
    else:
        max_consecutive_a = max(max_consecutive_a, consecutive_a)
        consecutive_a = 0

    # After 'b' must be 'i' or 't'
    if(txt[i] == 'b'):
        # 'b' is the last character or after 'b' is not 'i'
        if(i + 1 == len(txt) or txt[i + 1] not in ['i', 't']):
            if(i + 1 == len(txt)):
                print("no", i)
            else:
                print("no", i + 1)
            is_rabbit = False
            break

# Don't forget to find maximum consecutive once again
max_consecutive_a = max(max_consecutive_a, consecutive_a)

# Output
if(is_unknown):
    print("unknown", len(txt))
elif(is_rabbit):
    print("yes", max_consecutive_a)